### ISSUE TITLE
Continuous draining of the battery

### ISSUE THREAD

**Issue by elax46** [twrecked/hass-aarlo#895]
With the latest update I noticed a return to the battery draining of the cameras even if they are not used at all
![Screenshot 2024-03-03 130045](https://github.com/twrecked/hass-aarlo/assets/55278049/d4331a87-d0d7-4e44-ad29-d18dd93ba140)
This first screen is for a camera model with H7 hardware
![Screenshot 2024-03-03 130657](https://github.com/twrecked/hass-aarlo/assets/55278049/8a0f3c22-8a80-4f1b-8858-c8c45619b65c)
This secon screen is for a camera model with H8 hardware

**Comment by twrecked** [comment]
Which 2 versions did you go between?

**Comment by elax46** [comment]
0.7.4.2 i didn't use alpha version

**Comment by Avatar1976** [comment]
Sorry to do a 'me too' but my doorbell in particular ran itself flat after installation of v0.8.1.a12 . Not sure which version I was using prior but assume the latest prior to that. 
Will try downgrading version one back. 
![Screenshot_20240315_082125_Home Assistant](https://github.com/twrecked/hass-aarlo/assets/47725430/48f97f44-639f-47d5-9503-678937a503b8)

**Comment by chicknlil** [comment]
I noticed my almost empty battery before I saw I had camera support. When I added the card to the view, it was already streaming; my unscientific guess would be that something about these recent changes started the cameras streaming when they otherwise wouldn't have been.

I'm going to be charging my doorbell, and keep an eye on the streaming now.

**Comment by Avatar1976** [comment]
> I noticed my almost empty battery before I saw I had camera support. When I added the card to the view, it was already streaming; my unscientific guess would be that something about these recent changes started the cameras streaming when they otherwise wouldn't have been.
> 
> I'm going to be charging my doorbell, and keep an eye on the streaming now.

Yes your unscientific opinion / guess was what I was thinking from my experience too. Have been careful not to activate it / view it through HA but currently downgraded to .a11 also.

**Comment by chicknlil** [comment]
> 
> Yes your unscientific opinion / guess was what I was thinking from my experience too. Have been careful not to activate it / view it through HA but currently downgraded to .a11 also.

I already have automations to let me know when my keyless entry's (Eufy) battery drops below certain points so I can swap it out with a good one. I basically duplicated those automations but with the doorbell battery. So I'll get alerts on my phone, and when it's REALLY low, over all the house speakers.

Thing that really boggled me was that mine is a wireless that's also wired in, so it normally holds a charge rather than go up or down. To go down, and to go down like that was a WTF.

**Comment by twrecked** [comment]
I need more info.

- what version did you go from and to?
- what kind of doorbells? [Because my Essential doorbell is looking ok.]

And can you capture some debug logs? Especially around the startup.

Thanks

**Comment by chicknlil** [comment]
Mine actually drained before I did the update, oddly enough.

I was on .11, and upgraded (while I was charging the doorbell!) to .12.

I've got a Arlo Video Doorbell 2K (2nd Generation) - AVD4001 ... it's wireless, but I have it hardwired.

Walk me through giving you the debug logs and I happily will. 👍

**Comment by twrecked** [comment]
I've only just put in support for the gen2 doorbell - in release 12 - so it might not be correct. Let me check something first.

**Comment by twrecked** [comment]
I might need some more information. Can you go to `Developer Tools -> STATES` and look for the camera device of your doorbell.

I need the `Attributes` on the right. I don't need the sensitive informtion - the URLS and access code - but I'd like to see how the battery connections are being recorded. You can cut and paste it from the screen. 

For example,  my Essential Doorbell has the following info:

```
model_name: AVD2001A
brand: Arlo
frontend_stream_type: hls
battery_level: 94
battery_tech: Rechargeable
brightness: 0
flipped: false
mirrored: false
motion_detection_sensitivity: 80
signal_strength: 3
unseen_videos: 1382
recent_activity: false
image_source: capture/03-15 21:41
charging: false
charger_type: None
wired: false
wired_only: false
time_zone: America/Bogota
state: idle
attribution: Data provided by my.arlo.com
name: Front Door Bell
has_siren: true
device_brand: Arlo
device_model: AVD2001A
supported_features: 3
```

**Comment by chicknlil** [comment]
Thank you for the precise directions - I struggle sometimes to process "simple" tasks and that enabled me to find exactly what you were looking for!

```
model_name: AVD4001A
brand: Arlo
frontend_stream_type: hls
battery_level: 50
battery_tech: Rechargeable
brightness: 0
flipped: false
mirrored: false
motion_detection_sensitivity: 80
signal_strength: 3
unseen_videos: 733
recent_activity: false
image_source: capture/03-16 09:44
charging: true
charger_type: VAC Charger
wired: true
wired_only: false
time_zone: America/New_York
state: idle
attribution: Data provided by my.arlo.com
name: Doorbell
has_siren: true
device_brand: Arlo
device_model: AVD4001A
friendly_name: Doorbell
supported_features: 3
```

**Comment by twrecked** [comment]
Can you try the latest 0.8 release?

**Comment by chicknlil** [comment]
> Can you try the latest 0.8 release?

When I reconnected the doorbell to HA about an hour ago (when I saw your comment) and did the update, it was at 89%.  It's now at 88. I'll keep an eye on it.

**Comment by chicknlil** [comment]
> > Can you try the latest 0.8 release?
> 
> When I reconnected the doorbell to HA about an hour ago (when I saw your comment) and did the update, it was at 89%. It's now at 88. I'll keep an eye on it.

So far so good, still at 88%.

Will circle back tomorrow or if it drops suddenly, whichever comes first!

**Comment by Avatar1976** [comment]
Yeah happy to report charging gradually upward from my DC doorbell power supply. Updated to the latest version unhooked the doorbell (dead flat), plugged into usb to charge to 50%, unplugged then returned the doorbell to its mounting plate which trickle charges the doorbell. And you can see it gradually bringing the battery back up to nominal levels.
![Screenshot_20240318_113538_Home Assistant](https://github.com/twrecked/hass-aarlo/assets/47725430/14db6b45-e2aa-4ee9-8ea6-abe2bca0f415)

**Comment by chicknlil** [comment]
Mine is going down more than it was, but it's not _draining_.

![image](https://github.com/twrecked/hass-aarlo/assets/100124187/cb7d7faf-b90f-4bd5-abf2-347491a4fa0c)

Is there something I can tweak on my end with the polling for that particular device?

**Comment by twrecked** [comment]
@chicknlil Can you turn on debug logs. Add this to your `configuration.yaml` file then restart.

```yaml
logger:
  default: info
  logs:
    pyaarlo: debug
```

You just need to look for the phrases `NO ping` or `pinging` in the output. Similar to this but with the `XXXs` replaced with the device id.

```
2024-03-17 09:57:08,842 - pyaarlo - DEBUG - NO ping to XXXXXXXXXXXX
2024-03-17 09:57:08,843 - pyaarlo - DEBUG - XXXXXXXXXXX: pinging Rear Lights
```

For the doorbell there should be a `NO ping` entry.

**Comment by chicknlil** [comment]
> @chicknlil Can you turn on debug logs. Add this to your `configuration.yaml` file then restart.
> 
> ```yaml
> logger:
>   default: info
>   logs:
>     pyaarlo: debug
> ```
> 
> You just need to look for the phrases `NO ping` or `pinging` in the output. Similar to this but with the `XXXs` replaced with the device id.
> 
> ```
> 2024-03-17 09:57:08,842 - pyaarlo - DEBUG - NO ping to XXXXXXXXXXXX
> 2024-03-17 09:57:08,843 - pyaarlo - DEBUG - XXXXXXXXXXX: pinging Rear Lights
> ```
> 
> For the doorbell there should be a `NO ping` entry.

I added that to the configuration.yaml, but I am not finding anything like that in the logs unless I'm looking in the wrong spot.

**Comment by Avatar1976** [comment]
> Mine is going down more than it was, but it's not _draining_.
> 
> ![image](https://private-user-images.githubusercontent.com/100124187/313647966-cb7d7faf-b90f-4bd5-abf2-347491a4fa0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3OTU0MTcsIm5iZiI6MTcxMDc5NTExNywicGF0aCI6Ii8xMDAxMjQxODcvMzEzNjQ3OTY2LWNiN2Q3ZmFmLWI5MGYtNGJkNS1hYmYyLTM0NzQ5MWE0ZmEwYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxOFQyMDUxNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mOWZkMjJmYzQ3NmNmYjIyMTVmZjg2YTlmZWY4NGM0ZTdlMjcyMmZkM2ZiNGRkOWIzMGFlNjk1OWJjNTZhNDdiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aJT2GxyNqUzU_WCQxlbyKFBDZg_ddCS92ILd6U8wc4U)
> 
> Is there something I can tweak on my end with the polling for that particular device?

That's weird, are you definitely on the latest .08 (beta) release? My doorbell is exactly the same one as you and @twrecked 's fix worked flawlessly for me, DC voltage to your mounting plate unimpeded etc?
Just looking at the chunky way the battery drops there is it possible you have something still live streaming off it? It's like something is draining the battery but ha is only getting the updates on an event (ala no ping). By contrast the aarlo automation when pinging the doorbell draining it had it trending downward smoothly. Did you restart the doorbell in your arlo app after putting the newest release on.

**Comment by chicknlil** [comment]
>Did you restart the doorbell in your arlo app after putting the newest release on.

I didn't even know this was a thing. 😳

To be fair to me, I went from only having one Arlo camera in my carport) to it, the doorbell, and 5 more cameras in my condo. The "carport camera" always worked well enough, so I didn't poke too much!

Restarting **all** of my devices now and will see what happens over the next 24 hours!

**Comment by Avatar1976** [comment]
Just a FYI with doorbells in particularly it's meant to "seesaw" between 60% and 80% when hooked up via the classic 18v doorbell DC source. So you could be looking at normal behaviour here, took me a while to find that in the arlo documentation when I first got mine (why I only charged it to 50% in usb to confirm gradually battery recharging on doorbell voltage).

**Comment by chicknlil** [comment]
Mine is now sitting up at 79%.

I think things are all good on my end now too!

Good catch, @Avatar1976 !

**Comment by elax46** [comment]
I opened this problem request but I see that others have queued for the doorbell but what about the cameras? @twrecked

**Comment by Avatar1976** [comment]
> I opened this problem request but I see that others have queued for the doorbell 😅​😅​but what about the cameras? @twrecked

Maybe try the latest version? See what happens report back, from memory you were using a much older version as those alpha releases had been about for a while (most recent releases now changed back to beta).  It's hard to troubleshoot when you don't know if someone is working on your latest coding.

**Comment by elax46** [comment]
I understand your point but if it's called a beta channel I assume it's not a stable release and might introduce more problems rather than fix them. The latest version released to the official public is the one I installed

**Comment by Avatar1976** [comment]
> I understand your point but if it's called a beta channel I assume it's not a stable release and might introduce more problems rather than fix them. The latest version released to the official public is the one I installed

We're all beta in here 😂🤣
Yeah seriously though fair point. 😊

**Comment by twrecked** [comment]
@elax46 what are the model of camera you are seeing the issue with? I need the real model not the hardware revision.

**Comment by elax46** [comment]
> @elax46 what are the model of camera you are seeing the issue with? I need the real model not the hardware revision.

This is the first version with 3.7 Volt batteries

**Comment by twrecked** [comment]
Go to `Developer Tools -> STATES`. Search for the camera entity that is causing problems and can you tell me the model name. For example, one of mine is:

```
model_name: VMC4030
```

**Comment by elax46** [comment]
> Go to `Developer Tools -> STATES`. Search for the camera entity that is causing problems and can you tell me the model name. For example, one of mine is:
> 
> ```
> model_name: VMC4030
> ```

`model_name: VMC4030` and `VMC3030`

**Comment by twrecked** [comment]
I'll take a look at the code.

Interestingly, I have some VMC4030s here and they seem ok.
